### PR TITLE
Updating error message on ambiguous variable

### PR DESCRIPTION
**Pull request by yashcodes**
As keyword argument "category" no longer exist. Error message on ambiguous variable needs to be updated

Closes [#2035](https://github.com/jump-dev/JuMP.jl/issues/2035)

### CODE REVIEW

File: src/macros.jl [hunk 1]
```diff
@@ -1123,7 +1123,7 @@ macro variable(args...)
     else
         x = popfirst!(extra)
         if x in [:Int,:Bin,:PSD]
-            _error("Ambiguous variable name $x detected. Use the \"category\" keyword argument to specify a category for an anonymous variable.")
+            _error("Ambiguous variable name $x detected. To specify category of an anonymous variable use proper keyword argument or use appropiate function to add the required constraint on the variable.")
```
Comment (odow) [inline src/macros.jl:1126]: We should instead tell people to use `is_integer = true`, `is_binary = true` etc.

```Julia
if x == :Int
    _error(
        "Ambiguous variable name $x detected. To specify an anonymous integer " *
        "variable, use `@variable(model, is_integer = true)` instead."
    )
elseif == :Bin
    ...
```

**Comment by odow** [comment]
Tests? Coverage is a little confused.

**Comment by yashcodes** [comment]
If you want I can add tests but I don't think tests are important for these error message.

**Comment by yashcodes** [comment]
If you say I can add tests but I think tests are not important for these error messages.

**Comment by yashcodes** [comment]
Yeah cool I add tests.

**Comment by yashcodes** [comment]
Fine, I used `@test_macro_throws` :)

**Pull request merged**